TemplateManagerTools.sh

```diff
@@ -4,8 +4,8 @@ creator_icon="xapp-favorite-symbolic"
 editor_icon="edit"
 eraser_icon="delete"
 
-template_folder="$HOME/Templates"
-template_src_folder="$HOME/Templates/source"
+template_folder="$(xdg-user-dir TEMPLATES)" # Alternatively, place to one of the folders in kf5-config --path templates
```

Comment:
```suggestion
if [ "$(xdg-user-dir TEMPLATES)" != "$HOME" ]; then
    template_folder="$(xdg-user-dir TEMPLATES)"
else 
    IFS=':' read -ra template_possible_folders <<< "$(kf5-config --path templates)"
    counter=0
    unset template_folder
    while [ $counter -lt ${#template_possible_folders[@]} ] && [ ! -n "$template_folder" ]; do
        path="$(realpath "${template_possible_folders[$counter]}")"
        if [ "${path##$HOME}" != "${path}" ]; then
            template_folder="$path"
        fi
        ((counter++))
    done
fi
```
Really thanks for the suggestion.
To be honest, it was hardcoded because I haven't known xdg-user-dir command. I knew kf5-config, but it looked to me uncompleted as it doesn't include Template folder in home directory. With this xdg-user-dir, I can make a proper logic to always use user directory to avoid asking root permisons.

Reply:
Yeah, I also did not know about that, was surprised that `echo $XDG_TEMPLATES_DIR` returned nothing, than read https://wiki.archlinux.org/title/XDG_user_directories#Querying_configured_directories.

Also, I was surprised that in my `~/.config/user-dirs.dirs` the XDG_TEMPLATES_DIR was set to `$HOME`, but not to `$HOME/Templates`. Do you know if it is default in Arch?

Suggestions:
- instead of `! -n` use `-z` for better readability in if test statements
- inside template_src_folder variable use the template_folder variable to avoid running this code again for the source folder. Also, I am not sure if source folder by default should be with dot, i.e. `.source`, because in system path /usr/share/templates the desktop files store their templates in `/usr/share/templates/.source`.